### DISCUSSION TITLE
Migrations, snapshots and indexes

### DISCUSSION THREAD

**Discussion by onordgren** [directus/directus#11877] [Archive]
Hey!

Are there any recommendations on how to handle database updates between multiple environments? For example, I have directus running in staging, production and multiple local dev environments and would like to have a clean way of keeping them in sync. Read a bit about migrations and snapshots and to me it’s sounds like migrations might be the best way to move the database structure from dev to staging to production. Is this correct or is there a better way with snapshots?

Regarding indexes, is it in the road map to be able to add more indexes than just unique? And in the mean time, is the preferred way of adding them by doing a migration?

Thx a lot for a great CMS!

**Comment by rijkvanzanten** [comment]
The snapshots system is a great way to keep the database in sync between various projects. You can create a snapshot from the project you want to replicate the schema from, and apply it to any other project. This allows you to easily keep the data model in sync.

As for custom indexes, yes, the goal is to allow configuring indexes down the line as well. For now, you'll have to rely on custom migrations, or some other direct-to-database approach of configuring any database specific settings that Directus doesn't support (yet) 👍🏻

**Reply by onordgren** [reply]
I see. Thx for the answer!
